crates/config/src/zksync.rs

```diff
@@ -74,9 +74,11 @@ pub struct ZkSyncConfig {
     pub optimizer_details: Option<OptimizerDetails>,
 
     // zksolc suppressed warnings.
+    #[serde(deserialize_with = "deserialize_warning_set")]
```

Comment:
Where are we currently suppressing this error? I'd rather print the error at that point instead of hijacking `filter_map`.

Reply:
Not us directly, but figment is

Reply:
From what I tried, this is a way to avoid figment discarding the values because it uses `serde` behind the scenes. We could also validate it, but we would need to touch the foundry code where they define the load_config functions.

Reply:
How does foundry work around this limitation, when it fails to decode an `enum` config option? Does it also silently discard other settings?